docs/sql-programming-guide.md

```diff
@@ -1879,6 +1879,7 @@ working with timestamps in `pandas_udf`s to get the best performance, see
 
 ## Upgrading From Spark SQL 2.3 to 2.4
 
+  - In version 2.3 and earlier, if `spark.sql.caseSensitive` is set to true, then the `CURRENT_DATE` and `CURRENT_TIMESTAMP` functions incorrectly became case-sensitive and would resolve to columns (unless typed in lower case). In Spark 2.4 this has been fixed and the functions are no longer case-sensitive.
```

Comment:
Shall we give the specified version range like https://github.com/apache/spark/pull/21217/files#r203205223?

Reply:
I updated specifying the versions. As this is included in the migration guide to 2.4 and I don't think this commit will be backported to 2.3, I think it is fine to state 2.3, but it is good to mention that also 2.2.1 is affected. Thanks.